### ISSUE TITLE
Stack Overflow when running tasks that depend on `*`

### ISSUE THREAD

**Issue by rblaine95** [jdx/mise#3071]
**Describe the bug**
Using `mise@2024.11.14` I can run mise tasks without issue.
Upgrading to `mise@2024.11.15`, I suddenly get a Stack Overflow when running the same tasks.
```
$ mise tasks        
Name       Description     Source
hello:1    Hello from 1    ~/didx/mise/.mise.toml
hello:2    Hello from 2    ~/didx/mise/.mise.toml
hello:all  Hello from all  ~/didx/mise/.mise.toml

$ mise version
2024.11.15 macos-arm64 (d3d8dc0 2024-11-16)
$ mise run hello:all

thread 'main' has overflowed its stack
fatal runtime error: stack overflow
```

This seems to be related to a task that has `depends = ["hello:*"]` and then running that task.
Changing it to `depends = ["hello:1", "hello:2"]` seems to resolve the issue.
Manually downgrading to `2024.11.14` also resolves the issue.

**To Reproduce**
Create a task that has `depends = ["*"]` and then try to run it.
```toml
# .mise.toml
[tasks."hello:1"]
description = "Hello from 1"
run = "echo Hello from 1"

[tasks."hello:2"]
description = "Hello from 2"
run = "echo Hello from 2"

[tasks."hello:all"]
description = "Hello from all"
depends = ["hello:*"]
```

<img width="1624" alt="image" src="https://github.com/user-attachments/assets/5fbb081d-d5f9-4d4a-b0ee-c8a87d901320">

**Expected behavior**
Tasks should be run.

**`mise doctor` output**

```text
version: 2024.11.15 macos-arm64 (d3d8dc0 2024-11-16)
activated: yes
shims_on_path: no

build_info:
  Target: aarch64-apple-darwin
  Features: DEFAULT, NATIVE_TLS, OPENSSL
  Built: Sat, 16 Nov 2024 16:23:01 +0000
  Rust Version: rustc 1.82.0 (f6e511eec 2024-10-15)
  Profile: release

shell:
  /bin/zsh
  zsh 5.9 (arm64-apple-darwin24.0)

dirs:
  data: ~/.local/share/mise
  config: ~/.config/mise
  cache: ~/Library/Caches/mise
  state: ~/.local/state/mise
  shims: ~/.local/share/mise/shims

config_files:
  ~/.config/mise/config.toml
  ~/didx/mise/.mise.toml

backends:
  aqua
  asdf
  cargo
  core
  go
  npm
  pipx
  spm
  ubi
  vfox
  unknown

plugins:
  awscli      https://github.com/MetricMike/asdf-awscli.git#c26bbb4
  dotnet      https://github.com/hensou/asdf-dotnet#43d4a74
  helm        https://github.com/Antiarchitect/asdf-helm.git#091d00f
  helmfile    https://github.com/feniix/asdf-helmfile.git#818feac
  hurl        https://github.com/raimon49/asdf-hurl.git#473e2e9
  istioctl    https://github.com/virtualstaticvoid/asdf-istioctl.git#6ec43e8
  jq          https://github.com/mise-plugins/asdf-jq.git#6d86d19
  k6          https://github.com/gr1m0h/asdf-k6.git#107aa85
  kind        https://github.com/johnlayton/asdf-kind.git#1201d8c
  kubectl     https://github.com/asdf-community/asdf-kubectl.git#2fb3b57
  maven       https://github.com/mise-plugins/asdf-maven.git#c872214
  nsc         https://github.com/dex4er/asdf-nsc.git#4477407
  opentofu    https://github.com/virtualroot/asdf-opentofu.git#e11a7c8
  pnpm        https://github.com/jonathanmorley/asdf-pnpm.git#85221d3
  poetry      https://github.com/mise-plugins/mise-poetry.git#c1aea14
  pre-commit  https://github.com/jonathanmorley/asdf-pre-commit.git#26bfc42
  tilt        https://github.com/eaceaser/asdf-tilt.git#69eac7b
  usage       https://github.com/jdx/mise-usage.git#fe3888a
  uv          https://github.com/asdf-community/asdf-uv.git#8155fc5
  yq          https://github.com/sudermanjr/asdf-yq.git#772992f

toolset:
  ubi:jdx/usage@1.3.2
  aqua:jqlang/jq@1.7.1
  ubi:mikefarah/yq@4.44.5
  asdf:MetricMike/asdf-awscli@2.21.3

env_vars:
  MISE_SHELL=zsh

settings:
  activate_aggressive = false
  all_compile = false
  always_keep_download = false
  always_keep_install = false
  asdf_compat = false
  cache_prune_age = "30d"
  ci = false
  color = true
  debug = false
  disable_backends = []
  disable_default_registry = false
  disable_hints = []
  disable_tools = []
  experimental = true
  fetch_remote_versions_cache = "1h"
  fetch_remote_versions_timeout = "10s"
  go_default_packages_file = "~/.default-go-packages"
  go_download_mirror = "https://dl.google.com/go"
  go_repo = "https://github.com/golang/go"
  go_set_gopath = false
  go_set_goroot = true
  go_skip_checksum = false
  http_timeout = "30s"
  jobs = 4
  legacy_version_file = true
  legacy_version_file_disable_tools = []
  libgit2 = true
  lockfile = false
  log_level = "info"
  not_found_auto_install = true
  paranoid = false
  pin = false
  plugin_autoupdate_last_check_duration = "7d"
  python_compile = true
  quiet = false
  raw = false
  task_timings = false
  trace = false
  trusted_config_paths = []
  unix_default_file_shell_args = ["sh"]
  unix_default_inline_shell_args = [
      "sh",
      "-c",
  ]
  use_file_shell_for_executable_tasks = false
  use_versions_host = true
  verbose = false
  windows_default_file_shell_args = [
      "cmd",
      "/c",
  ]
  windows_default_inline_shell_args = [
      "cmd",
      "/c",
  ]
  windows_executable_extensions = [
      "exe",
      "bat",
      "cmd",
      "com",
      "ps1",
      "vbs",
  ]
  yes = false

  [cargo]
  binstall = true

  [node]

  [npm]
  bun = false

  [pipx]
  uvx = false

  [python]
  compile = true
  default_packages_file = "~/.default-python-packages"
  pyenv_repo = "https://github.com/pyenv/pyenv.git"
  venv_auto_create = false
  venv_stdlib = false

  [ruby]
  default_packages_file = "~/.default-gems"
  ruby_build_repo = "https://github.com/rbenv/ruby-build.git"
  ruby_install = false
  ruby_install_repo = "https://github.com/postmodern/ruby-install.git"

  [status]
  missing_tools = "if_other_versions_installed"
  show_env = false
  show_tools = false
No warnings found
No problems found

```

**Additional context**
```
$ mise run hello:all --trace
mise run hello:all --trace                                                                                                                                                              134 ↵
TRACE  1 [src/cli/mod.rs:224] run logger init
TRACE 10 [src/file.rs:161] cat ~/.local/share/mise/installs/go-github-com-google-go-containerregistry-cmd-crane/.mise.backend
TRACE 10 [src/file.rs:161] cat ~/.local/share/mise/installs/awscli/.mise.backend
TRACE  1 [src/eager.rs:27] post_settings
DEBUG  1 [src/cli/mod.rs:231] ARGS: /Users/robbie/.cargo/bin/mise run hello:all --trace
TRACE  1 [src/cli/mod.rs:232] run
TRACE  2 [src/backend/mod.rs:54] load_tools start
TRACE  1 [src/cli/run.rs:135] run init
TRACE  1 [src/config/mod.rs:81] load start
TRACE  2 [src/backend/mod.rs:57] load_tools core
TRACE 10 [src/file.rs:161] cat ~/.local/share/mise/installs/usage/.mise.backend
TRACE 10 [src/file.rs:161] cat ~/.local/share/mise/installs/go-github-com-nats-io-natscli-nats/.mise.backend
TRACE 10 [src/file.rs:161] cat ~/.local/share/mise/installs/npm-yarn/.mise.backend
TRACE 10 [src/file.rs:161] cat ~/.local/share/mise/installs/pipx-poetry/.mise.backend
TRACE 10 [src/toolset/install_state.rs:123] init_install_state tools
TRACE  2 [src/plugins/mise_plugin_toml.rs:29] parsing: /Users/robbie/.local/share/mise/plugins/poetry/mise.plugin.toml
TRACE  2 [src/file.rs:161] cat ~/.local/share/mise/plugins/poetry/mise.plugin.toml
TRACE  2 [src/backend/mod.rs:74] load_tools asdf
TRACE  2 [src/backend/mod.rs:86] load_tools vfox
TRACE  2 [src/plugins/mise_plugin_toml.rs:29] parsing: /Users/robbie/.local/share/mise/plugins/poetry/mise.plugin.toml
TRACE  2 [src/file.rs:161] cat ~/.local/share/mise/plugins/poetry/mise.plugin.toml
TRACE  2 [src/backend/mod.rs:96] load_tools install_state
TRACE  2 [src/backend/mod.rs:104] load_tools done
TRACE  5 [src/cache.rs:129] reading ~/Library/Caches/mise/maven/legacy_filenames-9389f.msgpack.z
TRACE 11 [src/cache.rs:129] reading ~/Library/Caches/mise/poetry/legacy_filenames-9389f.msgpack.z
TRACE  8 [src/cache.rs:129] reading ~/Library/Caches/mise/opentofu/legacy_filenames-9389f.msgpack.z
TRACE  1 [src/config/mod.rs:83] load legacy_files
TRACE  1 [src/config/mod.rs:89] load config_filenames
TRACE  1 [src/config/mod.rs:91] load config_paths
TRACE  1 [src/config/mod.rs:92] config_paths: ["/Users/robbie/didx/mise/.mise.toml", "/Users/robbie/.config/mise/config.toml"]
TRACE  2 [src/file.rs:161] cat ~/didx/mise/.mise.toml
TRACE  3 [src/file.rs:161] cat ~/.config/mise/config.toml
TRACE  2 [src/config/config_file/mise_toml.rs:92] parsing: ~/didx/mise/.mise.toml
TRACE  3 [src/config/config_file/mise_toml.rs:92] parsing: ~/.config/mise/config.toml
TRACE  1 [src/config/mod.rs:94] load config_files
TRACE  1 [src/config/mod.rs:103] load build
TRACE  1 [src/config/mod.rs:106] load validate
TRACE  1 [src/config/mod.rs:109] load all aliases
TRACE  1 [src/config/mod.rs:112] config: Config {
    Config Files: [
        "~/didx/mise/.mise.toml",
        "~/.config/mise/config.toml",
    ],
}
TRACE  1 [src/config/mod.rs:118] load done
TRACE  1 [src/config/mod.rs:277] load_all_tasks
TRACE  9 [src/file.rs:161] cat ~/didx/mise/.mise/tasks/hello/1
TRACE  5 [src/file.rs:161] cat ~/didx/mise/.mise/tasks/hello/2
TRACE  1 [src/config/mod.rs:295] loaded task hello:all – ~/didx/mise/.mise.toml
TRACE  1 [src/config/mod.rs:295] loaded task hello:2 – ~/didx/mise/.mise/tasks/hello/2
TRACE  1 [src/config/mod.rs:295] loaded task hello:1 – ~/didx/mise/.mise/tasks/hello/1
TRACE  1 [src/config/mod.rs:303] load_all_tasks 3
TRACE  1 [src/cli/run.rs:137] run get_task_lists
TRACE  1 [src/cli/run.rs:189] paralellize_tasks start
TRACE  1 [src/toolset/builder.rs:47] toolset::builder::build
TRACE  1 [src/config/mod.rs:528] load_env start
TRACE  1 [src/config/mod.rs:543] load_env done
TRACE  1 [src/config/mod.rs:545] EnvResults
TRACE  1 [src/toolset/mod.rs:482] env start
TRACE  1 [src/toolset/mod.rs:513] env end
TRACE  8 [src/cache.rs:129] reading ~/Library/Caches/mise/jq/1.7.1/list_bin_paths-9389f.msgpack.z
TRACE  5 [src/cache.rs:129] reading ~/Library/Caches/mise/usage/1.3.2/list_bin_paths-9389f.msgpack.z
TRACE  4 [src/cache.rs:129] reading ~/Library/Caches/mise/yq/4.44.5/list_bin_paths-9389f.msgpack.z
TRACE  6 [src/cache.rs:129] reading ~/Library/Caches/mise/awscli/2.21.3/list_bin_paths-9389f.msgpack.z

thread 'main' has overflowed its stack
fatal runtime error: stack overflow
```

**Comment by rblaine95** [comment]
This appears to have been resolved with the release of 2024.11.16 🚀

**Comment by rblaine95** [comment]
Apologies, I was too quick on the trigger
